WickedEngine/wiECS.h

```diff
@@ -346,6 +404,48 @@ namespace wi::ecs
 		// Disallow this to be copied by mistake
 		ComponentManager(const ComponentManager&) = delete;
 	};
+
+	// This is the class to store all ComponentManagers,
+	// this is useful for bulk operation of all attached components within an entity
+	class ComponentLibrary{
+	public:
+		uint64_t libraryVersion;
+		wi::vector<ComponentManager_Interface*> componentManagers;
+
+		//Create a Component Library and set up its version
+		ComponentLibrary(uint64_t iLibraryVersion = 0){ libraryVersion = iLibraryVersion; }
+
+		// Set library version, making sure different changes 
+		// between your component library are able to be serialized properly
+		void SetLibraryVersion(uint64_t& iLibraryVersion){ libraryVersion = iLibraryVersion; }
+
+		// Create an instance of ComponentManager of a certain data type
+		// Once added, cannot be removed!
+		template<typename T> inline ComponentManager<T>& Add(){
```

Comment:
Maybe it should be called `Register()` instead of Add, I think it would signal it better what is happening.